src/core/symbology/qgssymbol.cpp

```diff
@@ -1642,6 +1663,23 @@ void QgsSymbol::renderFeature( const QgsFeature &feature, QgsRenderContext &cont
     if ( needsExpressionContext )
       mSymbolRenderContext->expressionContextScope()->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "symbol_layer_index" ), symbolLayerIndex + 1, true ) );
 
+    // if this symbol layer has associated clip masks, we need to render it to a QPicture first so that we can
+    // determine the actual rendered bounds of the symbol. We'll then use that to retrieve the clip masks we need
+    // to apply when painting the symbol via this QPicture.
+    const bool hasClipGeometries = !context.testFlag( Qgis::RenderContextFlag::AlwaysUseGlobalMasks )
```

Comment:
Shall we not test `!mRenderHints.testFlag( Qgis::SymbolRenderHint::IsSymbolLayerSubSymbol` ? Isn't that the opposite of the condition above?

Reply:
Right, fixed